subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -201,6 +201,67 @@ class MavenPublishJavaIntegTest extends AbstractMavenPublishIntegTest {
         ]
     }
 
+    def "can publish java-library with dependency constraints"() {
```

Comment:
I guess we should also test this for publishing to Ivy repository. (IvyPublishJavaIntegTest)

Reply:
Yeah I guess so... 
I am adding these as part of #3630 to also treat the consumption in the tests.